iris-mpc/src/bin/server.rs

```diff
@@ -1012,6 +1016,66 @@ async fn server_main(config: Config) -> eyre::Result<()> {
     let db_chunks_bucket_name = config.db_chunks_bucket_name.clone();
     let db_chunks_folder_name = config.db_chunks_folder_name.clone();
 
+    // --------------------------------------------------------------------------
+    // ANCHOR: Syncing latest node state
+    // --------------------------------------------------------------------------
+    tracing::info!("⚓️ ANCHOR: Syncing latest node state");
+    let all_nodes = config.node_hostnames.clone();
+    let next_node = &all_nodes[(config.party_id + 1) % 3];
+    let prev_node = &all_nodes[(config.party_id + 2) % 3];
+
+    tracing::info!("Database store length is: {}", store_len);
+    let mut states = vec![my_state.clone()];
+    for host in [next_node, prev_node].iter() {
+        let res = reqwest::get(format!("http://{}:3000/sync", host)).await;
+        match res {
+            Ok(res) => {
+                let state: SyncState = match res.json().await {
+                    Ok(state) => state,
+                    Err(e) => {
+                        tracing::error!("Failed to parse sync state from party {}: {:?}", host, e);
+                        panic!(
+                            "could not get sync state from party {}, trying to restart",
+                            host
+                        );
+                    }
+                };
+                states.push(state);
+            }
+            Err(e) => {
+                tracing::error!("Failed to fetch sync state from party {}: {:?}", host, e);
+                panic!(
+                    "could not get sync state from party {}, trying to restart",
+                    host
```

Comment:
what should be the behavior on this? Since we already did a heartbeat once, this should be expected to always work, so restarting should be fine? Do we need some graceful shutdown already at this point?

Reply:
same as above?